clientconn.go

```diff
@@ -674,13 +680,27 @@ func (cc *ClientConn) Connect() {
 // context expires.  Returns nil unless the context expires first; otherwise
 // returns a status error based on the context.
 func (cc *ClientConn) waitForResolvedAddrs(ctx context.Context) error {
+	// Set the start time for name resolution if it's not already set.
+	cc.mu.Lock()
```

Comment:
This is called on every RPC.

You're unconditionally setting this flag and timestamp at the start of every RPC.  That can't be right.